mitodl/ol-django
# 2116: improve resilience of posthog integration

### What are the relevant tickets?
https://github.com/mitodl/mitxonline/issues/2116

### Description
Defines the generic Posthog functionality found in MITx Online and MIT Open to ol-django.  This ol-django functionality will later be imported into MIT Open and MITx Online in order to reduce redundant code across our applications.

### How can this be tested?
You can use this MITx Online branch to test the functionality provided by this PR: https://github.com/mitodl/mitxonline/tree/ol-django-posthog

#### Test setup
You will need to perform the following steps in order to test this PR using the MITx Online branch above.
1. From within the ol-django project, run `pants package "src/mitol/olposthog:"`
2. Copy the file from dist/mitol-django-olposthog-2024.4.16.tar.gz and paste it into the main directory of the MITx Online application.
3. From within the MITx Online application's root directory, run `docker-compose run -u root --rm web poetry add ./mitol-django-olposthog-2024.4.16.tar.gz`.
4. From the same location as step 3, run `docker-compose run -u root --rm web poetry install`.
5. From the same location as step 4, run `docker-compose build`.
6. Follow the "Installation and setup" section of the README included with this PR.
7. From within the MITx Online application's root directory, run `docker-compose up`.
8. Verify that your local MITx Online application appears as it should when `POSTHOG_ENABLED=True` is defined in your .env file.
9. Verify that your local MITx Online application appears as it should when `POSTHOG_ENABLED=False` is defined in your .env file.
